service/src/main/java/bio/terra/drshub/services/DrsProviderService.java

```diff
@@ -0,0 +1,99 @@
+package bio.terra.drshub.services;
+
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.drshub.config.DrsHubConfig;
+import bio.terra.drshub.config.DrsProvider;
+import java.net.URI;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.UriComponents;
+import org.springframework.web.util.UriComponentsBuilder;
+
+@Slf4j
+@Service
+public record DrsProviderService(DrsHubConfig drsHubConfig) {
+
+  private static final Pattern compactIdRegex =
+      Pattern.compile("(?:dos|drs)://(?<compactId>dg\\.[0-9a-z-]+)", Pattern.CASE_INSENSITIVE);
+
+  private static final Pattern hostNameRegex =
+      Pattern.compile("(?:dos|drs)://(?<hostname>[^?/]+\\.[^?/]+)", Pattern.CASE_INSENSITIVE);
```

Comment:
This is the remediation of the catastrophic backtracking regex bug

Reply:
I would like it to be a separate PR, but Sonarcloud flags it as a security issue and keeps me from merging the code, so its just kinda easier to do it here. We can easily change it back by just looking at the previous commit and making a hotfix.

Reply:
not 100% sure but I think this might work / simplify the query?
```suggestion
  private static final Pattern compactIdRegex =
      Pattern.compile("(?:dos|drs)://(?<compactId>dg\\.[0-9a-z-]+)", Pattern.CASE_INSENSITIVE);

  private static final Pattern hostNameRegex =
      Pattern.compile("(?:dos|drs)://(?<hostname>[^?/]+)", Pattern.CASE_INSENSITIVE);
```

Reply:
Unfortunately, no. The removal of `\\.` means we would match invalid hosts. It removes the check for at least one period, so we would match `drs://foo/<object_id>`, which is invalid. Needs to be at least `drs://foo.bar/<object_id>`.